### PR TITLE
feat: rate limit

### LINKED ISSUE #119: Add Rate-limiting

### PR DESCRIPTION
**Pull request by maxlivinci**
resolves #119

### CODE REVIEW

File: charts/murmurations/charts/index/templates/index/config.yaml [hunk 1]
```diff
@@ -12,3 +12,6 @@ data:
   ELASTICSEARCH_URL: http://index-es:9200
   NATS_CLUSTER_ID: murmurations
   NATS_URL: http://nats:4222
+  # Rate limit
+  # 10 reqs/minute: "10-M"
+  RATE_LIMIT_PERIOD: 10-M
```
Comment (maxlivinci) [inline charts/murmurations/charts/index/templates/index/config.yaml:17]: @geoffturk Feel free to change the value

File: charts/murmurations/charts/library/templates/library/config.yaml [hunk 2]
```diff
@@ -9,3 +9,6 @@ data:
   SERVER_TIMEOUT_IDLE: 15s
   MONGO_HOST: library-mongo:27017
   MONGO_DB_NAME: murmurationsLibrary
+  # Rate limit
+  # 10 reqs/minute: "10-M"
+  RATE_LIMIT_PERIOD: 10-M
```
Comment (maxlivinci) [inline charts/murmurations/charts/library/templates/library/config.yaml:14]: same here

**Comment by geoffturk** [comment]
@maxlivinci Is it possible to have more granular control over the rate limits? For example, it's OK to have a lot of reads (GETs) but we want to avoid a lot of writes (POSTs), so ideally we could control the limits for these two types of requests separately.

**Review by geoffturk — CHANGES_REQUESTED** [review]
See question above.

Also, can we return a JSON payload with an error description like we do for the 404?

image

image

**Comment by maxlivinci** [comment]
image

**Review by geoffturk — APPROVED** [review]
Perfect! LGTM 🚀

**Pull request merged**